src/date-range-picker/index.tsx

```diff
@@ -54,48 +54,35 @@ function renderDateRange({
   hideTimeOffset?: boolean;
   timeOffset: { startDate?: number; endDate?: number };
 }) {
-  if (!range) {
-    return (
-      <span className={styles['label-text']} aria-disabled={true}>
-        {placeholder}
-      </span>
-    );
-  }
-
-  const formatted =
-    range.type === 'relative' ? (
-      formatRelativeRange?.(range) ?? ''
-    ) : (
-      <BreakSpaces
-        text={formatDateRange({
-          startDate: range.startDate,
-          endDate: range.endDate,
-          timeOffset,
+  const firstPart = range
```

Comment:
I also tried the approach below where every span had a separate purpose, I found it a bit more elegant but unfortunately still crashed when changing an absolute value for a relative value. It seems like the content if the first inner span can't be set to an empty string:

```
  const formattedRelativeRange = (range?.type === 'relative' && formatRelativeRange?.(range)) ?? '';

  const formattedAbsoluteStartDate =
    range?.type === 'absolute'
      ? formatDateTimeWithOffset({
          date: range.startDate,
          timeOffset: timeOffset.startDate,
          hideTimeOffset,
          format: absoluteFormat,
          locale,
        })
      : '';

  const formattedAbsoluteEndDate =
    range?.type === 'absolute'
      ? formatDateTimeWithOffset({
          date: range.endDate,
          timeOffset: timeOffset.endDate,
          hideTimeOffset,
          format: absoluteFormat,
          locale,
        })
      : '';

  return (
    <span className={(!range && styles['label-text']) || undefined} aria-disabled={!range}>
      <span>{range?.type === 'relative' ? formattedRelativeRange : range ? '' : placeholder}</span>
      <span className={styles['label-token-nowrap']}>{formattedAbsoluteStartDate}</span>
      <span>{formattedAbsoluteEndDate && ' — '}</span>
      <span className={styles['label-token-nowrap']}>{formattedAbsoluteEndDate}</span>
    </span>
```